Modules/LearningModule/classes/class.ilStructureObjectGUI.php

```diff
@@ -583,7 +583,7 @@ public static function _goto(
         }
         
         if ($ilAccess->checkAccess("read", "", ROOT_FOLDER_ID)) {
-            ilUtil::sendFailure(sprintf(
+            $GLOBALS['DIC']->ui()->mainTemplate()->setOnScreenMessage('failure', sprintf(
```

Comment:
Another $GLOBALS introduction. This is a static method, but it already includes a global $DIC; at the beginning.